core/src/main/java/org/apache/iceberg/PartitionStatsUtil.java

```diff
@@ -87,13 +90,10 @@ private static PartitionMap<PartitionStats> collectStats(
       PartitionMap<PartitionStats> statsMap = PartitionMap.create(table.specs());
       int specId = manifest.partitionSpecId();
       PartitionSpec spec = table.specs().get(specId);
-      PartitionData keyTemplate = new PartitionData(partitionType);
 
       for (ManifestEntry<?> entry : reader.entries()) {
         ContentFile<?> file = entry.file();
-        StructLike coercedPartition =
-            PartitionUtil.coercePartition(partitionType, spec, file.partition());
-        StructLike key = keyTemplate.copyFor(coercedPartition);
+        Record key = coercedPartitionRecord(file, spec, partitionType);
```

Comment:
Need `Record` instead of `PartitionData` for the writers. 

Cannot keep this conversion in the data module as it just to wraps the same `PartitionStats` object.